autogluon/utils/tabular/ml/trainer/abstract_trainer.py

```diff
@@ -567,20 +568,26 @@ def stack_new_level_aux(self, X, y, level, fit=True, time_limit=None):
         X_train_stack_preds = self.get_inputs_to_stacker(X, level_start=0, level_end=level, fit=fit)
         return self.generate_weighted_ensemble(X=X_train_stack_preds, y=y, level=level, kfolds=0, n_repeats=1, stack_name=stack_name, time_limit=time_limit)
 
-    def generate_weighted_ensemble(self, X, y, level, kfolds=0, n_repeats=1, stack_name=None, hyperparameters=None, time_limit=None, base_model_names=None, name_suffix='', save_bagged_folds=None):
+    def generate_weighted_ensemble(self, X, y, level, kfolds=0, n_repeats=1, stack_name=None, hyperparameters=None, time_limit=None, base_model_names=None, name_suffix='', save_bagged_folds=None, check_if_best=True, child_hyperparameters=None):
         if save_bagged_folds is None:
             save_bagged_folds = self.save_bagged_folds
         if base_model_names is None:
             base_model_names = self.models_level['core'][level - 1]
         if len(base_model_names) == 0:
             logger.log(20, 'No base models to train on, skipping weighted ensemble...')
             return []
+
+        # TODO: Remove extra_params, currently a hack
+        if child_hyperparameters is not None:
+            extra_params = {'_tmp_greedy_hyperparameters': child_hyperparameters}
+        else:
+            extra_params = {}
         weighted_ensemble_model = WeightedEnsembleModel(path=self.path, name='weighted_ensemble' + name_suffix + '_k' + str(kfolds) + '_l' + str(level), base_model_names=base_model_names,
                                                         base_model_paths_dict=self.model_paths, base_model_types_dict=self.model_types, base_model_types_inner_dict=self.model_types_inner, base_model_performances_dict=self.model_performance, hyperparameters=hyperparameters,
-                                                        objective_func=self.objective_func, num_classes=self.num_classes, save_bagged_folds=save_bagged_folds, random_state=level+self.random_seed)
+                                                        objective_func=self.objective_func, num_classes=self.num_classes, save_bagged_folds=save_bagged_folds, random_state=level+self.random_seed, **extra_params)
 
         self.train_multi(X_train=X, y_train=y, X_test=None, y_test=None, models=[weighted_ensemble_model], kfolds=kfolds, n_repeats=n_repeats, hyperparameter_tune=False, feature_prune=False, stack_name=stack_name, level=level, time_limit=time_limit)
-        if weighted_ensemble_model.name in self.get_model_names_all():
+        if check_if_best and weighted_ensemble_model.name in self.get_model_names_all():
```

Comment:
BTW in future PR, may consider tie handling in the best_check and in GreedyEnsemble, I received some emails where user had validation-accuracy ties (I guess it's not that uncommon for limited data sizes)

Reply:
Agreed, I plan on reworking how best model is determined in future.